### PR TITLE
report of induced schema def attribs for #1238

### PR DESCRIPTION
**Pull request by turbomam**
- #1238

### CODE REVIEW

File: tests/test_issues/test_issue_1238.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+
+class SchemaDefInducedAttribsCase(unittest.TestCase):
+    def test_induced_attrib_list(self):
+        meta_view = SchemaView(META_URL)
+
+        induced_schema_def = meta_view.induced_class(SCRUTINIZED_CLASS)
+        induced_schema_def_attribs = induced_schema_def.attributes
+        induced_schema_def_attrib_names = [v.name for k, v in induced_schema_def_attribs.items()]
+        induced_schema_def_attrib_names.sort()
```
[the remark this answers is not in the document]
Comment (turbomam) [reply tests/test_issues/test_issue_1238.py:72]: Thanks. How about composition/list members equal?

File: tests/test_issues/test_issue_1238.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                            'todos',
+                            'types',
+                            'version']
+
+
+class SchemaDefInducedAttribsCase(unittest.TestCase):
+    def test_induced_attrib_list(self):
+        meta_view = SchemaView(META_URL)
+
+        induced_schema_def = meta_view.induced_class(SCRUTINIZED_CLASS)
```
Comment (cmungall) [inline tests/test_issues/test_issue_1238.py:69]: consider naming the variables to be consistent with the type. This sounds like it should hold a SchemaDefinition object, not a ClassDefinition object.

File: tests/test_issues/test_issue_1238.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        induced_schema_def_attribs = induced_schema_def.attributes
+        induced_schema_def_attrib_names = [v.name for k, v in induced_schema_def_attribs.items()]
```
Comment (cmungall) [inline tests/test_issues/test_issue_1238.py:71]: This is simpler:
`list(induced_schema_def_attribs.keys())`

File: tests/test_issues/test_issue_1238.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import unittest
+
+from linkml_runtime import SchemaView
+
+META_URL = "https://raw.githubusercontent.com/linkml/linkml-model/main/linkml_model/model/schema/meta.yaml"
```
Comment (cmungall) [inline tests/test_issues/test_issue_1238.py:5]: tests should be self contained and have no network dependencies

Additionally this test will break as soon as the metamodel changes and a new metaslot is added to ClassDefinition

Comment (turbomam) [reply tests/test_issues/test_issue_1238.py:5]: OK, I'm adding to this now but will ultimately remove the test.

If I did want to retain this test, is the metamodel available from linkml/linkml?

Also, what does it mean to add a metaslot to `ClassDefinition`, as opposed to a slot? Does "metaslot" emphasize that it's a slot from the metamodel, as opposed to a user schema? Is metaslot defined anywhere?

Comment (turbomam) [reply tests/test_issues/test_issue_1238.py:5]: Thanks for the review

Comment (turbomam) [reply tests/test_issues/test_issue_1238.py:5]: This was a starting point for 
- #1239

Comment (cmungall) [reply tests/test_issues/test_issue_1238.py:5]: > OK, I'm adding to this now but will ultimately remove the test.

I see. I encourage PRs for test cases but if the goal is to do entirely local testing best not to make a PR

> Also, what does it mean to add a metaslot to ClassDefinition, as opposed to a slot? Does "metaslot" emphasize that it's a slot from the metamodel, as opposed to a user schema? Is metaslot defined anywhere?

a metaslot is a slot in the metamodel; see 
https://linkml.io/linkml-model/docs/specification/03schemas/#schema-basics

**Comment by turbomam** [comment]
Whoa, this PR has great comments. But I don't have time to work though it any more now so will close.

**Pull request closed without merging**